addOns/commonlib/commonlib.gradle.kts

```diff
@@ -0,0 +1,128 @@
+import org.zaproxy.gradle.addon.AddOnStatus
+
+plugins {
+    `maven-publish`
+    signing
+}
+
+group = "org.zaproxy.addon"
+
+version = "0.1.0"
+description = "A common library, for use by other add-ons."
+
+zapAddOn {
+    addOnName.set("Common Library")
+    addOnStatus.set(AddOnStatus.ALPHA)
+    zapVersion.set("2.9.0")
+
+    manifest {
+        author.set("ZAP Dev Team")
+
+        helpSet {
+            baseName.set("help%LC%.helpset")
+            localeToken.set("%LC%")
+        }
+    }
+}
+
+dependencies {
+    zap("org.zaproxy:zap:2.7.0")
+
+    implementation("commons-io:commons-io:2.6")
+    implementation("org.apache.commons:commons-csv:1.8")
+    implementation("org.apache.commons:commons-collections4:4.4")
+
+    testImplementation(project(":testutils"))
+}
+
+
+val sourceSets = extensions.getByName("sourceSets") as SourceSetContainer
+
+tasks.register<Jar>("javadocJar") {
```

Comment:
Some classes lack JavaDoc, should we document all of them?

Reply:
I guess all the public methods at least.

I'm happy to do some of that after this is merged.

Reply:
There weren't that many, all public methods/classes should have JavaDoc also added the since tag.